### ISSUE TITLE
Fractions of a day

### ISSUE THREAD

**Issue by erapade** [Microsoft/powerbi-visuals-gantt#9]
As I can see, the Bar will start at not only the Year/Month/Day, but also the hour/minute/second defined with the value in StartDate. 
But the duration is always truncated to down to a integer (Whole Number)

Due to this, the bars are sometimes left empty as seen below
<img width="90" alt="capture" src="https://cloud.githubusercontent.com/assets/17324892/22951323/0d0b1912-f309-11e6-9e66-74bad6efcdde.PNG">

My data looks something like below:
<img width="206" alt="capture" src="https://cloud.githubusercontent.com/assets/17324892/22951702/6f84d85c-f30a-11e6-8c74-36e4d2bc5f06.PNG">
My suggestion is to allow and correctly handle fractions of a day, also for the Duration. 
I understand that in case of durations of just some minutes or seconds, it will be hard to see that time-slot, but that is the same problem as we have with other diarams, e.g. the Pie diagram

**Comment by erapade** [comment]
I looked in the code, but as I don't know Java Script, I will not manage this of my own. But I did see this line in gantt.js that probably is a type:
const MillisecondsInWeek: number = 4 * MillisecondsInADay;

Also: I have seen that duration is always at minimum one day. So if the task starts at 6 AM, it will always stop at 6 AM some days later. 

I have come soo far that I can re-compile the code, but I can't figure out what to change

**Comment by ignatvilesov** [comment]
Thanks for your feedback.
I'll address and fix this issue soon.

**Ignat Vilesov,**
Software Engineer

**Microsoft Power BI Custom Visuals**
pbicvsupport@microsoft.com

**Comment by erapade** [comment]
This still haven't been adressed. The week of 4 days still remains, as well as the ability to specify duration in shorter shuncks that one day. Please fix, the code change is very easy. For changing from 4 to 7 days in a week is also submitted as a pull request

**Comment by erapade** [comment]
I have done some updates [here](https://github.com/erapade/powerbi-visuals-gantt). I added new scales for Hours, Minutes, and Seconds as well. This now works fine for me, but it will need some fine tuning for a final release. 
- The scales/axis for Hours, Minutes, and Seconds needs to be handled better
- The resolution for duration when hovering needs to have better resolution than just one day
- The resolution for start time needs to have better resolution than just one day
- I have noticed that sometimes there's a 1 second glitch (white space between two bars). I have checked my start time, the calculated duration, and the next start time, and they are correct. This is not at all a big problem though
- I also noticed that using Resources does not perform well with the Grouping settings. The color will disappear. Maybe this is an issue for itself.
- Would also be nice to be able to use Date Hierarchies to drill up and down in time scale, but that's out of my capacity, and maybe requires a new issue/branch as well

I have added example data below. In this case there's a 1 second glitch 29 seconds from start
I use the following mapping
